test/Microsoft.ML.Tests/ImagesTests.cs

```diff
@@ -710,5 +711,59 @@ public void TestBackAndForthConversionWithoutAlphaNoInterleaveNoOffset()
                 Done();
             }
         }
+
+        [Fact]
+        public void ImageResizerTransformResizingModeFill()
+        {
+            var env = new MLContext();
+            var dataFile = GetDataPath("images/fillmode.tsv");
+            var imageFolder = Path.GetDirectoryName(dataFile);
+            var data = TextLoader.Create(env, new TextLoader.Arguments()
+            {
+                Columns = new[]
+                {
+                    new TextLoader.Column("ImagePath", DataKind.TX, 0)
+                }
+            }, new MultiFileSource(dataFile));
+
+            const int targetDimension = 50;
+            var pipe = new ImageLoadingEstimator(env, imageFolder, ("ImageReal", "ImagePath"))
+                .Append(new ImageResizingEstimator(env, "ImageReal", targetDimension, targetDimension, "ImageReal",
+                    resizing: ImageResizerTransformer.ResizingKind.Fill));
+
+            var rowView = pipe.Preview(data).RowView;
+            Assert.Single(rowView);
+
+            using (var bitmap = (Bitmap) rowView.First().Values.Last().Value)
+            {
+                // these points must be white
+                var topLeft = bitmap.GetPixel(0, 0);
+                var topRight = bitmap.GetPixel(bitmap.Width - 1, 0);
+                var bottomLeft = bitmap.GetPixel(0, bitmap.Height - 1);
+                var bottomRight = bitmap.GetPixel(bitmap.Width - 1, bitmap.Height - 1);
+                var middle = bitmap.GetPixel(bitmap.Width / 2, bitmap.Height / 2);
+
+                // these points must be red
+                var midTop = bitmap.GetPixel(bitmap.Width / 2, bitmap.Height / 3);
+                var midBottom = bitmap.GetPixel(bitmap.Width / 2, bitmap.Height / 3 * 2);
+                var leftMid = bitmap.GetPixel(bitmap.Width / 3, bitmap.Height / 2);
+                var rightMid = bitmap.GetPixel(bitmap.Width / 3 * 2, bitmap.Height / 2);
+
+                Assert.All(new[] { topLeft, topRight, bottomLeft, bottomRight, middle }, c =>
```

Comment:
>  Assert.All(new[] { topLeft, topRight, bottomLeft, bottomRight, middle }, c => [](start = 14, length = 79)

The joy of multi environment support. Mac OS and Linux fails because some of values become 254 instead of 255.

Reply:
Hm, weird. Some kind of rounding error, I guess. I'll rewrite the test to be a little more permissive.